crates/provider-wit-bindgen-macro/src/bindgen_visitor.rs

```diff
@@ -0,0 +1,800 @@
+                            let (_arg_name, converted_tokens) = convert_to_owned_type_arg(
+                                &self.serde_extended_structs,
+                                &self.type_lookup,
+                                fn_arg,
+                                self.replace_witified_maps,
+                            );
+
+                            let updated_fn_arg = syn::parse2::<FnArg>(converted_tokens)
+                                .expect("failed to produce valid FnArg from owned type conversion");
+
+                            trace!(
+                                "fn_arg [{}] converted to [{}]",
+                                fn_arg.to_token_stream().to_string(),
+                                updated_fn_arg.to_token_stream().to_string()
+                            );
+
+                            *fn_arg = updated_fn_arg;
+                        }
+
+                        // Get the return type of the function, we expect it to be a wasmtime::Result<T>
+                        let output_type = trimmed_fn
+                            .sig
+                            .output
+                            .to_token_stream()
+                            .into_iter()
+                            .collect::<Vec<TokenTree>>();
+
+                        // If we're dealing with a wasmtime::Result<T> type as we expect then we should pull out the T
+                        // so we can use them standalone as a return type
+                        let inner_tokens = match extract_wasmtime_result_type(&output_type) {
+                            Some(t) => t,
+                            None => {
+                                debug!("skipping function [{fn_name}] in impl [{impl_type_name}] which does not return a wasmtime::Result");
+                                continue;
+                            }
+                        };
+
+                        // Modify the functions output type
+                        trimmed_fn.sig.output =
+                            syn::parse2::<ReturnType>(quote::quote!(-> #inner_tokens))
+                                .expect("failed to purge wasmtime::Result from method return");
+
+                        // Save the manipulated function signature as a method, which will be put in a trait
+                        // that the provider must implement
+                        let full_path = self.generate_export_path();
+                        debug!(
+                                "adding exported trait method for path [{full_path}], trimmed signature: [{}]",
+                                trimmed_fn.sig.to_token_stream().to_string(),
+                            );
+                        self.export_trait_methods
+                            .entry(full_path)
+                            .or_default()
+                            .push(trimmed_fn);
+                    }
+                }
+            }
```

Comment:
@connorsmith256 this is the most important bit -- the structure of generated code changed, and where the methods show up and how they needed to be processed had to be adjusted. This code was previously dealing primarily with`Item::Struct`s